### DISCUSSION TITLE
How to prevent nxml-mode-hook from running in the exporter

### DISCUSSION THREAD

**Discussion by QiangF** [kjambunathan/org-mode-ox-odt#204] [Q&A] [answered]
I have put pprint-xml in nxml-mode-hook, it slows down the exporter terribley. 

    (setup nxml
       (:hook my-pprint-xml)
       (:autoload my-pprint-xml)
       (:when-loaded
         (defun my-pprint-xml ()
           (interactive)
           (read-only-mode -1)
           (pprint-xml (point-min) (point-max)))

         (defun my-xml-format ()
           "Format an XML buffer with `xmllint' (sudo apt-get install libxml2-utils)."
           (interactive)
           (shell-command-on-region (point-min) (point-max)
                                    "xmllint -format -"
                                    (current-buffer) t
                                    "*Xmllint Error Buffer*" t))

         (defun pprint-xml (begin end)
           "Pretty format XML markup in region. You need to have nxml-mode
         http://www.emacswiki.org/cgi-bin/wiki/NxmlMode installed to do
         this. The function inserts linebreaks to separate tags that have
         nothing but whitespace between them. It then indents the markup
         by using nxml's indentation rules."
           (interactive "r")
           (save-excursion
             (nxml-mode)
             (goto-char begin)
             (while (search-forward-regexp "\>[ \\t]*\<" nil t)
               (backward-char) (insert "\n") (setq end (1+ end)))
             (indent-region begin end)))))

**Comment by kjambunathan** [comment, marked as answer]
If you want pretty printed xml,


1. Use LibreOffice to resave.  See the bottom of this Wiki link https://github.com/kjambunathan/org-mode-ox-odt/wiki

2. Install `tidy`, and add

    ```
    #+odt_prettify_xml: tidy
    ```

These two options are sufficient for most debugging.  Running `tidy`
etc on XML code will introduce unwanted whitespaces.
